File: design-patterns/go/chain_of_responsibility/cache/button_ui.go
// button_ui.go

package main

import "fmt"

type ButtonUi struct {
	name string
}

func NewButtonUi(name string) (rcvr *ButtonUi) {
	rcvr = &ButtonUi{}
	rcvr.name = name
	return
}

func (rcvr *ButtonUi) Print() {
	fmt.Println(fmt.Sprintf("%v%v%v", "Printing ", rcvr.name, " Button"))
}

func (rcvr *ButtonUi) Remove() {
	fmt.Println(fmt.Sprintf("%v%v%v", "Removing ", rcvr.name, "  Button"))
}

File: design-patterns/ts/command/ui-element/button-ui.ts
// button-ui.ts

import UiCommand from "./ui-command";

class ButtonUi implements UiCommand {
    private name: string;
    constructor(name: string) {
        this.name = name;
    }
    public print() {
        console.log("Printing " + this.name + " Button");
    }
    public remove() {
        console.log("Removing " + this.name + "  Button");
    }
}

export default ButtonUi;